pkcs11/src/new/functions/object_management.rs

```diff
@@ -211,4 +126,54 @@ impl Pkcs11 {
             Rv::Error(rv_error) => Err(rv_error.into()),
         }
     }
+
+    // Ignore the unavailable one. One has to call the get_attribute_info method to check which
+    // ones are unavailable.
+    pub fn get_attributes(
+        &self,
+        session: &Session,
+        object: ObjectHandle,
+        attributes: &[AttributeType],
+    ) -> Result<Vec<Attribute>> {
+        let attrs_info = self.get_attribute_info(session, object, attributes)?;
+
+        // Allocating a chunk of memory where to put the attributes value.
+        let attrs_memory: Vec<(AttributeType, Vec<u8>)> = attrs_info
+            .iter()
+            .zip(attributes.iter())
+            .filter_map(|(attr_info, attr_type)| {
+                if let AttributeInfo::Available(size) = attr_info {
+                    Some((*attr_type, vec![0; *size]))
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        let mut template: Vec<CK_ATTRIBUTE> = attrs_memory
+            .iter()
+            .map(|(attr_type, memory)| {
+                Ok(CK_ATTRIBUTE {
+                    type_: (*attr_type).into(),
+                    pValue: memory.as_ptr() as *mut std::ffi::c_void,
+                    ulValueLen: memory.len().try_into()?,
+                })
+            })
+            .collect::<Result<Vec<CK_ATTRIBUTE>>>()?;
+
+        // This should only return OK as all attributes asked should be
+        // available. Concurrency problem?
```

Comment:
I meant "Concurrency Problem" if one of the attribute is made unavailable in between the two `C_GetAttributeValue` calls with the same session. Maybe this should be one of the function that needs to execute in a critical section.

Reply:
is it possible for an attribute to go AWOL?

Reply:
I think it's possible that someone modifies an object to set it private or maybe even delete it